grpc/federation/cel/url.go

```diff
@@ -0,0 +1,611 @@
+					v, err := url.Parse(string(args[0].(types.String)))
+					if err != nil {
+						return types.NewErr(err.Error())
+					}
+					return lib.toURLValue(*v)
+				},
+			),
+		),
+		BindFunction(
+			createURLName("parseRequestURI"),
+			OverloadFunc(createURLID("parseRequestURI_string_url"), []*cel.Type{cel.StringType}, URLType,
+				func(_ context.Context, args ...ref.Val) ref.Val {
+					v, err := url.ParseRequestURI(string(args[0].(types.String)))
+					if err != nil {
+						return types.NewErr(err.Error())
+					}
+					return lib.toURLValue(*v)
+				},
+			),
+		),
+		BindMemberFunction(
+			"scheme",
+			MemberOverloadFunc(createURLID("scheme_url_string"), URLType, []*cel.Type{}, cel.StringType,
+				func(_ context.Context, self ref.Val, args ...ref.Val) ref.Val {
+					v, err := lib.refToGoURLValue(self)
+					if err != nil {
+						return types.NewErr(err.Error())
+					}
+					return types.String(v.Scheme)
+				},
+			),
+		),
+		BindMemberFunction(
+			"opaque",
+			MemberOverloadFunc(createURLID("opaque_url_string"), URLType, []*cel.Type{}, cel.StringType,
+				func(_ context.Context, self ref.Val, args ...ref.Val) ref.Val {
+					v, err := self.Value().(*URL).GoURL()
+					if err != nil {
+						return types.NewErr(err.Error())
+					}
+					return types.String(v.Opaque)
+				},
+			),
+		),
+		BindMemberFunction(
+			"userinfo", // user is not a valid field name
+			MemberOverloadFunc(createURLID("user_url_userinfo"), URLType, []*cel.Type{}, UserinfoType,
+				func(_ context.Context, self ref.Val, args ...ref.Val) ref.Val {
+					v, err := self.Value().(*URL).GoURL()
+					if err != nil {
+						return types.NewErr(err.Error())
+					}
+					password, hasPassword := v.User.Password()
+					return lib.toUserinfoValue(v.User.Username(), password, hasPassword)
+				},
+			),
+		),
+		BindMemberFunction(
+			"host",
+			MemberOverloadFunc(createURLID("host_url_string"), URLType, []*cel.Type{}, cel.StringType,
+				func(_ context.Context, self ref.Val, args ...ref.Val) ref.Val {
+					v, err := self.Value().(*URL).GoURL()
+					if err != nil {
+						return types.NewErr(err.Error())
+					}
+					return types.String(v.Host)
+				},
+			),
+		),
+		BindMemberFunction(
+			"path",
+			MemberOverloadFunc(createURLID("path_url_string"), URLType, []*cel.Type{}, cel.StringType,
+				func(_ context.Context, self ref.Val, args ...ref.Val) ref.Val {
+					v, err := self.Value().(*URL).GoURL()
+					if err != nil {
+						return types.NewErr(err.Error())
+					}
+					return types.String(v.Path)
+				},
+			),
+		),
+		BindMemberFunction(
+			"rawPath",
+			MemberOverloadFunc(createURLID("rawPath_url_string"), URLType, []*cel.Type{}, cel.StringType,
+				func(_ context.Context, self ref.Val, args ...ref.Val) ref.Val {
+					v, err := self.Value().(*URL).GoURL()
+					if err != nil {
+						return types.NewErr(err.Error())
+					}
+					return types.String(v.RawPath)
+				},
+			),
+		),
+		// forceQuery is changed to return String
```

Comment:
Why is it? 🤔 Would you add a comment to it?

Reply:
I misread the implementation. I have corrected it.